### PR TITLE
Allow HTTP client to optionally send a payload for DELETE requests

### PR DESCRIPTION
**Pull request by bdragon**
Closes [#3736](https://github.com/wunderteam/portal/issues/3736).

### CODE REVIEW

File: lib/synapse_pay_rest/http_client.rb [hunk 1]
```diff
@@ -123,8 +123,10 @@ def get(path)
     # @raise [SynapsePayRest::Error] subclass depends on HTTP response
     # 
     # @return [Hash] API response
-    def delete(path)
-      response = with_error_handling { RestClient.delete(full_url(path), headers) }
+    def delete(path, payload = nil)
+      execute_arguments = { :method => :delete, :url => full_url(path), :headers => headers, :timeout => 300 }
```
Comment (Alex-Quinn) [inline lib/synapse_pay_rest/http_client.rb:136]: I don't think it's problematic but why was this timeout option added?

File: lib/synapse_pay_rest/http_client.rb [hunk 2]
```diff
@@ -123,8 +123,10 @@ def get(path)
     # @raise [SynapsePayRest::Error] subclass depends on HTTP response
     # 
     # @return [Hash] API response
```
Comment (Alex-Quinn) [inline lib/synapse_pay_rest/http_client.rb:134]: It probably makes sense to add the new `payload` arg to the doc comment. I'm mostly thinking in the context of potentially getting this merged into the main repo.

**Review by Alex-Quinn — APPROVED** [review]
I'm not sure if you're actually planning on merging this but it looks good to me.

🍡

File: lib/synapse_pay_rest/http_client.rb (hunk 1 above)
Comment (bdragon) [reply lib/synapse_pay_rest/http_client.rb:136]: I just followed the behavior in the other client methods that call `execute` directly

**Pull request merged**